framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -214,9 +212,14 @@ private void viewpointAdaptExecutable(
                 Element varargsElement =
                         methodCalled.getParameters().get(methodCalled.getParameters().size() - 1);
                 TypeMirror tm = ElementUtils.getType(varargsElement);
-                argReceivers.add(
-                        new FlowExpressions.ArrayCreation(
-                                tm, Collections.emptyList(), initializers));
+                if (initializers.size() == 1 && initializers.get(0).getType().equals(tm)) {
```

Comment:
This is a bad way to check if the argument passed was an array. It works considering that an exception would have been thrown if it was an invalid type.